webpack.config.js

```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+const path = require('path');
+const Htmlwebpackplugin = require('html-webpack-plugin');
+
+module.exports = {
```

Comment:
We need to use a plugin to delete the public folder before every build. We also want 4 different files all in the config folder. We want a `webpack.plugins.js` file which will export all our initialized plugins. We want a `webpack.config.js` file for our configuration that is the same for both development and production and that will be merged in the respective files using  `webpack-merge`. We want a `webpack.dev.js` file for development alone which will include the configuration for the dev server. Finally, we want a `webpack.prod.js` file that will have js uglification (or not since webpack uses Terser for production build by default so we can skip this), post-css processing (adding auto-prefixers for different browsers) and css optimization. Please fix, thanks.